File: java/src/Intersection_349.java
import java.util.*;
import java.util.stream.Collectors;

public class Intersection_349 {

    public int[] intersection(int[] nums1, int[] nums2) {

        Set<Integer> map1 = Arrays.stream(nums1).boxed().collect(Collectors.toSet());

        Set<Integer> set = new HashSet<>();
        for (int j : nums2) {
            if (!map1.contains(j)) continue;
            set.add(j);
        }

        int[] res = new int[set.size()];
        int i = 0;
        for (int num : set) {
            res[i++] = num;
        }
        return res;
    }
}

File: go/classic/Intersection_349.go
package classic

func intersection(nums1 []int, nums2 []int) []int {
	memo := make(map[int]int)

	for _, num := range nums1 {
		memo[num] = 1
	}

	map4nums2 := make(map[int]int)

	for _, num := range nums2 {
		if _, ok := memo[num]; ok {
			map4nums2[num] = 1
		}
	}

	res := make([]int, 0)

	for key, _ := range map4nums2 {
		res = append(res, key)
	}
	return res
}
